examples/erc721_events_decoded.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+/*
+ This is a fun example of decoding ERC721 events with the built-in decoder.  We use the Clovers.network contracts as an example, which are tokens that are minted by players who work out a symetric othelo game. View more at https://clovers.network
+*/
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	hypersyncgo "github.com/enviodev/hypersync-client-go"
+	"github.com/enviodev/hypersync-client-go/contracts"
+	"github.com/enviodev/hypersync-client-go/decoder"
+	"github.com/enviodev/hypersync-client-go/logger"
+	"github.com/enviodev/hypersync-client-go/options"
+	"github.com/enviodev/hypersync-client-go/types"
+	"github.com/enviodev/hypersync-client-go/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"go.uber.org/zap"
+)
+
+func main() {
+	opts := options.Options{
+		Blockchains: []options.Node{
+			{
+				Type:        utils.EthereumNetwork,
+				NetworkId:   utils.EthereumNetworkID,
+				Endpoint:    "https://eth.hypersync.xyz",
+				RpcEndpoint: "https://eth.rpc.hypersync.xyz",
+			},
+		},
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	hsClient, err := hypersyncgo.NewHyper(ctx, opts)
+	if err != nil {
+		logger.L().Error("failed to create hyper client", zap.Error(err))
+		return
+	}
+
+	client, found := hsClient.GetClient(utils.EthereumNetworkID)
+	if !found {
+		logger.L().Error("failure to discover hyper client", zap.Error(err))
+		return
+	}
+
+	startBlock := big.NewInt(20000000)
+	endBlock := big.NewInt(20865800)
+
+	logger.L().Info("Fetching USDC transfer logs", zap.Any("start_block", startBlock), zap.Any("end_block", endBlock))
+
+	// Clovers Transfer event signature: Transfer(address,address,uint256)
+	selections := []types.LogSelection{
+		{
+			Address: []common.Address{common.HexToAddress("0xB55C5cAc5014C662fDBF21A2C59Cd45403C482Fd")}, // Clovers contract address
+			Topics: [][]common.Hash{
+				{common.HexToHash("0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef")}, // Transfer event signature
+			},
```

Comment:
Do I have to create a new selection for every log type that I want to get from a contract?  What if I wanted to get the `ApprovalForAll` event also from this contract?

Reply:
In this specific case yes. We didn't implement loading contracts and abis to figure out available events for contracts nor do we have storage of these information yet in go. This is just a basic very fast example.

Reply:
For this abi that's provided in the contracts, events needs to be extracted from it and can be passed here to automatically get all of the ERC-721 events.